src/parser.ts

```diff
@@ -196,10 +207,28 @@ function parseNonLiteral(
     case 'UNNAMED_SCHEMA':
       return set(newInterface(schema as SchemaSchema, options, rootSchema, processed, usedNames, keyName, keyNameFromDefinition))
     case 'UNTYPED_ARRAY':
+      const {minItems, maxItems} = schema
+      const params = T_ANY
+      if ((typeof minItems === 'number' && minItems > 0) ||
+        (typeof maxItems === 'number' && maxItems > 0)) {
```

Comment:
If you accept my previous suggestion of adding another normalizer step, it could simplify this logic.